### PR TITLE
Webservices

### PR DESCRIPTION
**Pull request by VIBHAV19**
Not completed will continue by Friday

### CODE REVIEW

File: src/exercise1.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import re
+from flask import Flask
+
+app = Flask(__name__)
+
+@app.route('/email/<string:mail>', methods = ['GET'])
```
Comment (preshly) [inline src/exercise1.py]: The email should not be passed via the url parameters.
It can be passed in the api body

File: src/exercise1.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+def validate(mail):
+    valid = r'\b[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Z|a-z]{2,7}\b'
+    if re.match(valid,mail):
+        return "Valid"
+    return "Invalid"
+
```
Comment (preshly) [inline src/exercise1.py:15]: Please test the edge case
mail = a@a.com@a.com

File: src/exercise2.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import re
+from flask import Flask
+
+app = Flask(__name__)
+
+@app.route('/password/<string:password>', methods = ['GET'])
+def validate(password):
```
Comment (preshly) [inline src/exercise2.py]: The password should not be passed via the url parameters.
It can be passed in the api body

File: src/exercise4.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+  if item_resp == []:
+    return Response("{'error': 'Item id may not be present in the database.'}", mimetype='application/json', status=500)
+  return Response(json.dumps(item_resp), mimetype='application/json', status=200)
+
+
+@app.route('/savetoexcel/<string:filename>', methods = ['GET'])
+def save_data(filename):
+  item_resp = item_actions.save_data(filename)
+  if item_resp == []:
+    return Response("{'error': 'Item id may not be present in the database.'}", mimetype='application/json', status=500)
```
Comment (preshly) [inline src/exercise4.py]: message can be improved when there is an error as per the case

File: src/exercise4.py [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+  return Response(json.dumps(item_resp), mimetype='application/json', status=200)
+
+
+
+@app.route('/item', methods = ['POST'])
+def delete_item():
+  request_data = request.get_json()
+  id = request_data['id']
+  item_resp = item_actions.delete_item(id)
+  print(item_resp)
```
Comment (preshly) [inline src/exercise4.py]: print statements can be ignored

File: src/item_actions.py [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+
+  def add_item(self, item, reminder):
+    try:
+      item = self.item_repo.add_item(item, reminder)
+      return item
+    except Exception as e:
+      print(e)
+      return {}
+
+  def save_data(self, filename):
```
Comment (preshly) [inline src/item_actions.py]: function name can be improved

File: src/item_repository.py [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+    try:
+      self.connect_db()
+      cursor = self.connection.cursor()
+      cursor.execute('select count(id) from items where id = ?',(item,))
+      val = cursor.fetchone()[0]
+      if  val ==0 :
+        return {}
+      delete_cursor = cursor.execute('delete from items where id = ?', (item,))
+      self.connection.commit()
+      return "Delete was successful"
```
Comment (preshly) [inline src/item_repository.py]: the function can be refactored.
It is performing two sql statements, but the function name is just `delete_item`

File: src/item_repository.py [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+    try:
+      self.connect_db()
+      cursor = self.connection.cursor()
+      cursor.execute('select count(id) from items where id = ?',(id,))
+      val = cursor.fetchone()[0]
+      if  val ==0 :
+        return {}
+      update_cursor = cursor.execute('update items set item =?,status=?,reminder=?  where id = ?', (item,status,reminder,id))
+      self.connection.commit()
+      return "Update was successful"
```
Comment (preshly) [inline src/item_repository.py]: function can be refactored to update specific item properties

File: src/item_repository.py [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+        return "No rows to be added to the file"
+      column_names = list()
+      for i in cursor.description:
+          column_names.append(i[0])
+      result.append(column_names)
+      for row in rows:
+        result.append(row)
+      for row in result:
+        file.writerow(row)
+      return "Writting was successfully"
```
Comment (preshly) [inline src/item_repository.py:130]: the logic in the function can be moved to actions

File: src/main.py [hunk 10]
```diff
@@ -0,0 +1,27 @@
+
```
Comment (preshly) [inline src/main.py]: is this file needed?

**Review by preshly — CHANGES_REQUESTED** [review]
Please follow the coding conventions and syntax consistent at all places.

More unit and API tests can be added.

-- Vibhav : Ok will add soon

File: src/exercise1.py (hunk 1 above)
Comment (VIBHAV19) [reply src/exercise1.py]: Done

File: src/exercise1.py (hunk 2 above)
Comment (VIBHAV19) [reply src/exercise1.py:15]: Done

File: src/item_repository.py (hunk 9 above)
Comment (VIBHAV19) [reply src/item_repository.py:130]: Didn't understand this comment

File: src/main.py (hunk 10 above)
Comment (VIBHAV19) [reply src/main.py]: The file main.py has been deleted

**Review by preshly — APPROVED** [review]
Please move all the contents inside a folder, so that the changes could be merge to the main branch.
